### PR TITLE
Fixes #197 - Changes application structure approach

### LINKED ISSUE #197: Custom layer is reinitialised on each request
This is an example of the application with the custom layer `ConfigurationEnvironment` provided. 

```scala
object EntryPoint extends ZLambda[V2HttpEvent, V2HTTPResponse] { self =>

  override def apply(event: V2HttpEvent, context: Context): Task[V2HTTPResponse] =
    app(event,context).provideLayer(ConfigurationEnvironment.live)


  private def app(event: V2HttpEvent, context: Context) =
    for {
      _ <- ZIO.unit
    } yield V2HTTPResponse(200,Map(),Map(),List(), base64str,true)

}
```

This custom layer may contain heavy initialization logic that is expensive to execute on each request. 
But this is exactly what is happening with `ConfigurationEnvironment` because the layer is provided outside the app initialization logic. This way the application can't benefit from a warmed container. 

There must be a way to inject a layer to the code that initializes the whole lambda function process logic and propagate this layer to the `apply` method or something like this.

### PR DESCRIPTION
**Pull request by heaven-born**
This MR introduced a new zio-http-like application structure in order to tackle issue with  ZIO custom layer reinitialization on each request. 

Now the lambda app can be created like this:

```scala
object SimpleHandler extends ZIOAppDefault {

  val app = (event: CustomEvent, _: Context) =>
    for {
      _ <- printLine(event.message)
    } yield "Handler ran successfully"


override val run =
  ZLambdaRunner.serve(app)
```

@jrmsamson  some doc compilation task that seems to be not related to my fix fails for some reason. Not sure why. But you can check the general approach. I also tested it on my local app on `native-image` except fro `AWS Custom Layer`, not sure how to test it.

btw, running `sbt docs/generateReadme` does nothing on my local machine. It says:
```
info: Compiling 3 files to /Users/[user.name]/IdeaProjects/zio-lambda/zio-lambda-docs/target/website/docs
info: Compiled in 0.12s (0 errors)
[info] The new README.md file generated
```
but doesn't provide anything new to push. Some issues with the project version.

### CODE REVIEW

File: lambda/src/main/scala/zio/lambda/ZLambda.scala
Comment (heaven-born) [inline lambda/src/main/scala/zio/lambda/ZLambda.scala]: looks not very pleasant, but this is the only place where I can get the types.

**Comment by jrmsamson** [comment]
Sure I can help you with the testing

**Comment by heaven-born** [comment]
@jrmsamson I couldn't get how to test AWS Lambda Layers, but the rest of the code seems ready for review.
 
Not much of documentation changes were needed. The same for tests.

**Comment by jrmsamson** [comment]
I tested it as a AWS Custom Layer and it works like a charm. 

Have two comments about these changes:

1) Would be great if ZLambdaRunner.default layer is provided automatically, so users don't need to do it themselves

2) Some unit tests for this new approach would be great.

Apart from these two points. Looks good to me, good job.

**Comment by heaven-born** [comment]
Regarding your points:

 1. done.
 2. I'm not sure what I can test in the context of the "approach".  `ZLambdaRunner` is not testable because it includes default layers now, and there is not much code inside to test. Do you think `zio.lambda.internal.LoopProcessor.Live.loopZioApp`  shuld be tested?  It has minimal changes in comparison to the original `loop`, but at least it can be tested.

**Review by jrmsamson — APPROVED** [review]
LGTM good job. 👍🏽 

Please before merging make sure to update https://github.com/zio/zio-lambda/blob/master/build.sbt#L61 as it's pointing to the wrong class as you renamed it.

**Comment by heaven-born** [comment]
done. 

But it looks like I can't merge even approved PRs to the repo without write access.

**Pull request merged**
